src/app/home.cljs

```diff
@@ -1,14 +1,36 @@
 (ns app.home
   (:require [app.data :as data]
-            [app.state :as state]))
+            [app.state :as state]
+            [app.problem :as problem]
+            [reagent.core :as r]))
+
+(defonce sort-by-status (r/atom true))
+
+(defn sorted-problems []
+  (let [solved-keys (set (keys @problem/user-data))
+        p-map (group-by #(contains? solved-keys (:_id %)) data/problems)]
+    (if @sort-by-status
+      (concat (get p-map true) (get p-map false))
+      (concat (get p-map false) (get p-map true)))))
```

Comment:
I think grouping like this will have an unintended consequence that problems will never be sorted by id.

One way to do this would be to just create a new problem list with the user solution data assoc'd in. 

Something like this: `(map #(assoc % :solution (get solutions (:_id %))) data/problems)` Then a `sort-by` could work.

This also gives us the benefit that we have the final data already present. We will no longer have to look-up user solutions in `get-problem-status`.